acoular/sources.py

```diff
@@ -521,59 +748,117 @@ def result(self, num=128):
+    >>> # Create a meshgrid for 3D coordinates, with z=0 for all points
+    >>> X, Y = np.meshgrid(x, y)
+    >>> Z = np.zeros_like(X)  # Set all z-values to 0
+    >>>
+    >>> # Stack the coordinates into a single (3,9) array
+    >>> points = np.vstack([X.ravel(), Y.ravel(), Z.ravel()])
+    >>> points
+    array([[-1.,  0.,  1., -1.,  0.,  1., -1.,  0.,  1.],
+           [-1., -1., -1.,  0.,  0.,  0.,  1.,  1.,  1.],
+           [ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.]])
+
+    Now, to set the actual point source (``ps``), we define a microphone geomerity (``mg``), using
+    the positional data from ``points``, and a sine generator (``sg``) with a total number of 6
+    samples.
+
+    >>> from acoular import PointSource, SineGenerator, MicGeom
+    >>> mg = MicGeom(pos_total=points)
+    >>> sg = SineGenerator(freq=1000, sample_freq=51200, num_samples=6)
+    >>> ps = PointSource(signal=sg, loc=(0.5, 0.5, 1.0), mics=mg)
+
+    We choose a blocksize of 4 and generate the output signal at the microphones in blocks:
+
+    >>> for block in ps.result(num=4):
+    ...     print(block.shape)
+    (4, 9)
+    (2, 9)
+
+    The first block has shape (4,9) for 4 samples and 9 microphones. The second block has shape
+    (2,9), since of a total of 6 samples only 2 remained.
     """
 
-    #:  Emitted signal, instance of the :class:`~acoular.signals.SignalGenerator` class.
+    #: Instance of the :class:`~acoular.signals.SignalGenerator` class defining the emitted signal.
     signal = Instance(SignalGenerator)
 
-    #: Location of source in (`x`, `y`, `z`) coordinates (left-oriented system).
+    #: Coordinates ``(x, y, z)`` of the source in a left-oriented system. Default is
+    #: ``(0.0, 0.0, 1.0)``.
     loc = Tuple((0.0, 0.0, 1.0), desc='source location')
 
-    #: Number of channels in output, is set automatically /
-    #: depends on used microphone geometry.
+    #: Number of output channels, automatically set based on the :attr:`microphone geometry<mics>`.
     num_channels = Delegate('mics', 'num_mics')
 
-    #: :class:`~acoular.microphones.MicGeom` object that provides the microphone locations.
+    #: :class:`~acoular.microphones.MicGeom` object defining the positions of the microphones.
     mics = Instance(MicGeom, desc='microphone geometry')
 
     def _validate_locations(self):
         dist = self.env._r(array(self.loc).reshape((3, 1)), self.mics.pos)
         if npany(dist < 1e-7):
             warn('Source and microphone locations are identical.', Warning, stacklevel=2)
 
-    #: :class:`~acoular.environments.Environment` or derived object,
-    #: which provides information about the sound propagation in the medium.
+    #: An :class:`~acoular.environments.Environment` or derived object providing sound propagation
+    #: details, such as :attr:`speed of sound in the medium<acoular.environments.Environment.c>`.
+    #: Default is :class:`~acoular.environments.Environment`.
     env = Instance(Environment(), Environment)
```

Comment:
This is still rendered weirdly. I wonder why. I think this

```suggestion
    env = Instance(Environment)
```
would work. However, I don't really understand the difference when reading the docs https://docs.enthought.com/traits/traits_user_manual/defining.html#instance

@adku1173 @esarradj Are both equivalent?

Reply:
No, it is not equivalent. The Environment object should be set here as a default. However, the current implementation could be improved by lazily instantiating the environment object.

```suggestion
    env = Instance(Environment, args=())
```